pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -38,7 +46,108 @@ func (srv PrometheusSrv) RouteGetAlertStatuses(c *models.ReqContext) response.Re
 }
 
 func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Response {
-	recipient := c.Params(":Recipient")
-	srv.log.Info("RouteGetRuleStatuses: ", "Recipient", recipient)
-	return response.Error(http.StatusNotImplemented, "", nil)
+	ruleResponse := apimodels.RuleResponse{
+		DiscoveryBase: apimodels.DiscoveryBase{
+			Status: "success",
+		},
+		Data: apimodels.RuleDiscovery{},
+	}
+
+	ruleGroupQuery := ngmodels.ListOrgRuleGroupsQuery{
+		OrgID: c.SignedInUser.OrgId,
+	}
+	if err := srv.store.GetOrgRuleGroups(&ruleGroupQuery); err != nil {
+		ruleResponse.DiscoveryBase.Status = "error"
+		ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rule groups: %s", err.Error())
+		ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+		return response.JSON(http.StatusInternalServerError, ruleResponse)
+	}
+
+	for _, groupId := range ruleGroupQuery.Result {
+		alertRuleQuery := ngmodels.ListRuleGroupAlertRulesQuery{OrgID: c.SignedInUser.OrgId, RuleGroup: groupId}
+		if err := srv.store.GetRuleGroupAlertRules(&alertRuleQuery); err != nil {
+			ruleResponse.DiscoveryBase.Status = "error"
+			ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting rules for group %s: %s", groupId, err.Error())
+			ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+			return response.JSON(http.StatusInternalServerError, ruleResponse)
+		}
+
+		newGroup := &apimodels.RuleGroup{
+			Name:           groupId,
+			File:           "", // TODO: set this once we are setting it somewhere
+			LastEvaluation: time.Time{},
+			EvaluationTime: 0, // TODO: see if we are able to pass this along with evaluation results
+		}
+		for _, rule := range alertRuleQuery.Result {
+			instanceQuery := ngmodels.ListAlertInstancesQuery{
+				DefinitionOrgID: c.SignedInUser.OrgId,
+				DefinitionUID:   rule.UID,
+			}
+			if err := srv.store.ListAlertInstances(&instanceQuery); err != nil {
+				ruleResponse.DiscoveryBase.Status = "error"
+				ruleResponse.DiscoveryBase.Error = fmt.Sprintf("failure getting alerts for rule %s: %s", rule.UID, err.Error())
+				ruleResponse.DiscoveryBase.ErrorType = apiv1.ErrServer
+				return response.JSON(http.StatusInternalServerError, ruleResponse)
+			}
+
+			alertingRule := apimodels.AlertingRule{
+				State:       "inactive",
+				Name:        rule.Title,
+				Query:       "", // TODO: get this from parsing AlertRule.Data
+				Duration:    time.Duration(rule.For).Seconds(),
+				Annotations: rule.Annotations,
+			}
+
+			newRule := apimodels.Rule{
+				Name:           rule.Title,
+				Labels:         nil,                    // TODO: NG AlertRule does not have labels but does have annotations
```

Comment:
it will have labels though eventually, right?

Reply:
I am not sure on this. We can add it as a feature, but I believe at least for the translation of Dashboard alerts what currently are "tags" in the dasboard alerts I think are for only for notifications. So therefore I think the Tags in the migration become annotations, so it is not a requirement of the migration I don't believe.

That being said, I don't think it is a hard feature to add.

Reply:
My understanding is that we need labels for MVP, because routing & silencing will work on labels. I'll add this topic to next sync then

Reply:
So far only alert (instances) have labels not the rules. We discussed it with @kylebrandt recently and we are not sure we need labels for rules.
Do you have a use case in mind where we need them?

Reply:
Let's try and reach a decision about whether and when to add labels at the sync. This endpoint should have no trouble finding them wherever they are so they can be added in an additional PR. I'll leave the TODO in as a reminder to come back to this.